src/m365/spo/commands/site/site-archive.spec.ts

```diff
@@ -0,0 +1,192 @@
+            "_ObjectIdentity_": "3bf525a1-202d-8000-b136-71cce6ed75ac|908bed80-a04a-4433-b4a0-883d9847d110:1e852b49-bf4b-4ba5-bcd4-a8c4706c8ed4\nSpoOperation\nArchiveSite\n638505831432627104\nhttps%3a%2f%2fcontoso.sharepoint.com%2fsites%2fSales\n00000000-0000-0000-0000-000000000000",
+            "HasTimedout": false,
+            "IsComplete": true,
+            "PollingInterval": 15000
+          }
+        ]);
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(cli.promptForConfirmation);
+    sinon.stub(cli, 'promptForConfirmation').resolves(true);
+    await command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/Sales' } });
+    assert(postStub.called);
+  });
+
+  it('archives the site without prompting for confirmation when force option specified', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf(`/_vti_bin/client.svc/ProcessQuery`) > -1 &&
+        opts.data === `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="2" ObjectPathId="1" /><ObjectPath Id="4" ObjectPathId="3" /><Query Id="5" ObjectPathId="3"><Query SelectAllProperties="true"><Properties /></Query></Query></Actions><ObjectPaths><Constructor Id="1" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /><Method Id="3" ParentId="1" Name="ArchiveSiteByUrl"><Parameters><Parameter Type="String">https://contoso.sharepoint.com/sites/Sales</Parameter></Parameters></Method></ObjectPaths></Request>`) {
+        return JSON.stringify([
+          {
+            "SchemaVersion": "15.0.0.0", "LibraryVersion": "16.0.24817.12008", "ErrorInfo": null, "TraceCorrelationId": "3bf525a1-202d-8000-b136-71cce6ed75ac"
+          }, 2, {
+            "IsNull": false
+          },
+          4,
+          {
+            "IsNull": false
+          },
+          5,
+          {
+            "_ObjectType_": "Microsoft.Online.SharePoint.TenantAdministration.SpoOperation",
+            "_ObjectIdentity_": "3bf525a1-202d-8000-b136-71cce6ed75ac|908bed80-a04a-4433-b4a0-883d9847d110:1e852b49-bf4b-4ba5-bcd4-a8c4706c8ed4\nSpoOperation\nArchiveSite\n638505831432627104\nhttps%3a%2f%2fcontoso.sharepoint.com%2fsites%2fSales\n00000000-0000-0000-0000-000000000000",
+            "HasTimedout": false,
+            "IsComplete": true,
+            "PollingInterval": 15000
+          }
+        ]);
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/Sales', force: true, verbose: true } });
+    assert(loggerLogSpy.notCalled);
+  });
+
+  it('correctly handles API error', async () => {
+    sinon.stub(request, 'post').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf(`/_vti_bin/client.svc/ProcessQuery`) > -1) {
```

Comment:
In this case, it's not really needed to check for a URL. Since we just want to test if the message was logged correctfully.
So let's use `sinon.stub(request, 'post').rejects({errorObject})`.

Reply:
I have removed the URL check from the test. I am using `callsFake` to resolve with a response that contains an `ErrorInfo` object to test this part:
```
        if (response.ErrorInfo) {
          throw response.ErrorInfo.ErrorMessage;
        }
```

Reply:
Yes, it's entirely fine to add a test for this. But in this test we don't really care if we passed the right URL (that's already tested in other tests). In this test we only care that if we catch an error, that it's properly displayed.
Therefore I think we can remove the URL check.